### PR TITLE
feat(inputs.postgresql_extensible): introduce MaxVersion

### LINKED ISSUE #13568: postgres_extensible: allow specifying version upper bound
### Use Case

PostgreSQL 13 renamed fields in `pg_stat_statements` view: `total_time` - > `total_exec_time`.  There is already a way to specify minimal version to run query on, but it is not enough to handle this case. Upper version bound should be specified as well.

### Expected behavior

Maybe add a "semver" field  which accepts a string as a semver constraint? Something like "< 13.0" and deprecate `version` (which should be called `minversion`) in favour of it. Semver constraint can express arbitrary version constraints which should cover all the cases

### Actual behavior

-

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by redbaron**
# Required for all PRs

- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #13568

Rename `version` to `minversion` and introduce `maxversion` to limit DB version query should run on.

### CODE REVIEW

**Comment by redbaron** [comment]
I can't run tests on my M2 mac, something about docker timing out

File: plugins/inputs/postgresql_extensible/postgresql_extensible.go [hunk 1]
```diff
@@ -37,7 +37,9 @@ type Postgresql struct {
 type query []struct {
 	Sqlquery    string
 	Script      string
-	Version     int
+	Version     int `deprecated:"1.28.0;use minVersion to specify minimal DB version this query supports"`
```
Comment (srebhan) [inline plugins/inputs/postgresql_extensible/postgresql_extensible.go:40]:
```suggestion
	Version     int `deprecated:"1.28.0;use 'min_version' instead"`
```

File: plugins/inputs/postgresql_extensible/postgresql_extensible.go [hunk 2]
```diff
@@ -37,7 +37,9 @@ type Postgresql struct {
...
+	MinVersion  int
+	MaxVersion  int
```
Comment (srebhan) [inline plugins/inputs/postgresql_extensible/postgresql_extensible.go:42]: Please add TOML tags for those options!

File: plugins/inputs/postgresql_extensible/sample.conf [hunk 3]
```diff
@@ -35,20 +35,30 @@
...
+  # a single integer without last component, for example:
+  # 9.6.2 -> 906
+  # 15.2 -> 1500
+  #
   # Structure :
   # [[inputs.postgresql_extensible.query]]
   #   sqlquery string
-  #   version string
+  #   minversion int
+  #   maxversion int
```
Comment (srebhan) [inline plugins/inputs/postgresql_extensible/sample.conf:53]: By default TOML will name them `min_version` and `max_version` if I'm not mistaken...

**Review by srebhan — COMMENTED** [review]
Thanks for the PR @redbaron! Just some minor comments...

**Comment by redbaron** [comment]
>  avoid the integration tests as there is where a bunch of docker containers are launched.

Thing is, this plugin tests require PostgreSQL container

**Comment by redbaron** [comment]
Integration test failure seems to be unrelated

**Comment by powersj** [comment]
> I am running from VSCode, which internally close to `go test` I think

Correct vscode will run go test without the `-short` option. I only suggest the `-short` option so you can continue to run at least the unit tests. There appears to be an issue with go1.20.6 + [go-testcontainers](https://github.com/testcontainers/testcontainers-go/issues/1306) + [docker](https://github.com/moby/moby/pull/45942) right now.

**Review by powersj — APPROVED** [review]
Thank you!

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for the nice feature @redbaron!

**Pull request merged**